src/esp/sim/Simulator.h

```diff
@@ -716,6 +727,16 @@ class Simulator {
                              scene::SceneNode& agentParentNode);
   agent::Agent::ptr addAgent(const agent::AgentConfiguration& agentConfig);
 
+  /**
+   * @brief Initialize sensor and attach to sceneNode of a particular object
+   * with uuid objectName_objectId
+   * @param objectId    Id of the object to which a sensor will be initialized
+   * at its node
+   * @param objectName Name of the object which will be used in sensor's uuid
+   *
+   */
+  void addSensorToObject(const int objectId, const std::string& objectName);
```

Comment:
This seems like a super specific method to have on the simulator class.  It's nice for a demo, but because it takes just an ID and a name, it isn't generally useful because the user has no control over the sensors position or orientation with moving the object.

Reply:
Agreed, this could be a nice utility (possibly in python utils?) given an optional transformation. You can create a new child SceneNode and apply the transformation there. Also think about what happens to the Sensor when the object is deleted.

Reply:
Thanks, that makes a lot of sense - I'll look into adding an optional transformation and making sure to delete the sensor when the object is deleted. 

I see that there is a method Simulator::setTransformation(const Magnum::Matrix4& transform, const int objectID, const int sceneID). Since the sensor is attached to the node at objectID, the sensor will move as well upon transformation, but I imagine that we would want another optional transform so that we could have a 3rd person view of an object as well! Am I understanding this right?

I think a function like this is needed to provide add a sensor to an object, but maybe this is not the best place. Would you recommend moving this to another class?